### PR TITLE
New Relic Doc

### PR DESCRIPTION
**Pull request by jdmooney**
First draft of New Relic doc

### CODE REVIEW

File: userguide/tutorials/metrics-newrelic.adoc [hunk 1]
```diff
@@ -0,0 +1,172 @@
+= Managing Metrics with New Relic
+
+:imagesdir: https://github.com/killbill/killbill-docs/raw/v3/userguide/assets/aws
+
+
+== Single Tier and Multitier Implementations
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]: Please change "single tier" to "single-tier" throughout. Same for "multitier" and "multi-tier." I've already made these changes for most of our documents. (The Datadog one got reverted when I went through an epic battle with git/GitHub recently, but I'll fix that.)

File: userguide/tutorials/metrics-newrelic.adoc [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+Measurements and metrics are an important part of managing your Kill
+Bill system to be sure that performance is optimized and any problems
+can be detected early. Kill Bill collects over 1,300 metrics to show you
+every conceivable view of your system's behavior.
```
Comment (FlowWriter) [inline userguide/tutorials/metrics-newrelic.adoc:11]: I'm still curious how users access a list of what these metrics are. @pierre showed me where they were, but said we shouldn't publicly announce it. Do users (clients) just ask us privately, then? Or do they dig around themselves until they find it?

File: userguide/tutorials/metrics-newrelic.adoc [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+Several third party tools are available to manage and display metrics of
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
Several third-party tools are available to manage and display metrics of
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+this type. One popular tool is https://www.newrelic.com/[New Relic]. This
+package offers a low-cost option to monitor and display whatever metrics
+you may find useful from the Kill Bill collection, including metrics
+that show the performance and resource use of Kill Bill and Kaui
+themselves, and metrics describing the infrastructure on which these
+systems run.
+
+New Relic can be integrated with several types of Kill Bill installations. In this guide we discuss the use of new relic to manage metrics produced
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
New Relic can be integrated with several types of Kill Bill installations. In this guide we discuss the use of New Relic to manage metrics produced
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+New Relic and work with them in the way that best suits your needs.
+
+== Check Your Kill Bill Version
+
+We assume that you have a working Kill Bill implementation running on
+AWS, using either the
+https://docs.killbill.io/latest/aws-singletier.html[Single Tier] or
+https://docs.killbill.io/latest/aws-multitier.html[Multitier]
+installation options. The single tier option runs Kill Bill, Kaui, the
+MariaDB database and the nginx load balancer on a single AWS EC2
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
MariaDB database, and the nginx load balancer on a single AWS EC2
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+installing the *New Relic Agent* in the system producing the data.
+
+1. In the main New Relic page, select the button labeled *Add Some Data*. If this button does not appear, select *+ Add More Data* in the upper right corner. 
+2. An overlay pane should appear. At the top of this pane select *Guided Install*.
+3. The next pane is headed *Installation Plan*. Make sure the *Auto-Discovery* tab is selected, then choose *Linux*.
+4. The next pane summarizes what you have selected. Click on *Begin Installation*. You will see the following page:
+
+image::metrics-newrelic-installation.png[align=center]
+
+5. In the box containing the install command, click *Copy Command*. The command will be copied to your clipboard.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
[start=5]
. In the box containing the install command, click *Copy Command*. The command will be copied to your clipboard.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+[NOTE]
+This command includes the API key assigned to you by
+New Relic. This authorizes the agent to do its job. This is your *user* API Key, and should not be confused with the key you will use in the next step.
+
+To install the agent:
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
[start=6]
. Install the agent: 
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+[NOTE]
+This command includes the API key assigned to you by
+New Relic. This authorizes the agent to do its job. This is your *user* API Key, and should not be confused with the key you will use in the next step.
+
+To install the agent: 
+
+1. Login to your EC2 instance. 
+2. Paste the saved install command to the Ubuntu command line. 
+3. Press Enter. The installation should begin.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
.. Login to your EC2 instance. 
.. Paste the saved install command to the Ubuntu command line. 
.. Press Enter. The installation should begin.
```

I think this ^^ should render as sub-steps a, b, and c.

File: userguide/tutorials/metrics-newrelic.adoc [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+You will be presented with a list of options to install, as shown below. An additional option, Infrastructure Agent, is always installed. Press Enter to select all of the options.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
You will be presented with a list of options to install, as shown below. An additional option, Infrastructure Agent, is always installed. 

[start=7]
. Press Enter to select all of the options.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+image::metrics-newrelic-options.png[align=center]
+
+
+You will be presented with several questions including default answers. Press Enter after each question to select the defaults.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
You will be presented with several questions including default answers. 

[start=8]
. Press Enter after each question to select the defaults.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+Finally, you will be asked for the MySQL password. Type `root`.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
Finally, you will be asked for the MySQL password. 

[start=9]
. Type `root`.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+
+If all goes well, each installation should report `success`.
+
+
+If you are using the Multitier installation, you will have multiple EC2
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
[NOTE]
*Note:* If you are using the Multitier installation, you will have multiple EC2
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+instances. The agent should be installed separately for each one.
+
+To test the installation, return to the main page. Select *Browse Data* from the top menu, then select *Metrics* from the dropdown.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
== Test the Installation

. Return to the main page. 
. Select *Browse Data* from the top menu. 
. Select *Metrics* from the dropdown.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+You should see a
+long list of metrics! These are coming from your instances, but they
+only describe the Ubuntu environment itself. You do not yet have the
+Kill Bill metrics. To fix this, proceed to the next step.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
Kill Bill metrics. To fix this, proceed to the next section.
```

File: userguide/tutorials/metrics-newrelic.adoc [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+
+The API key to be added is called the *Insights Insert Key.* This is *not* the same as the user API Key contained in the command you
+used to install the agent.
+
+To find the Insights Insert Key:
+
+1. Close any overlay panes on the New Relic page by clicking on the `X` in the upper right corner.
+2. Click the symbol at the far upper right to open your account menu, then select *API keys*.
+3. On the page that appears, select *Insights Insert keys* at the lower right.
+4. In the *Insert Keys* column on the left, under the word *key* at the bottom click *Show*.
```
Comment (ghost) [inline userguide/tutorials/metrics-newrelic.adoc]:
```suggestion
4. In the *Insert Keys* column on the left, under the word *key* at the bottom, click *Show*.
```

**Review by ghost — COMMENTED** [review]
Reviewed!

File: userguide/tutorials/metrics-newrelic.adoc (hunk 1 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 3 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 4 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 6 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 5 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 7 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 8 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: Fine. I did not know how to force step numbers and substeps.

File: userguide/tutorials/metrics-newrelic.adoc (hunk 9 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: OK

File: userguide/tutorials/metrics-newrelic.adoc (hunk 10 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 11 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 12 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: fixed, including multi-tier

File: userguide/tutorials/metrics-newrelic.adoc (hunk 13 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 14 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

File: userguide/tutorials/metrics-newrelic.adoc (hunk 15 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc]: done

**Comment by jdmooney** [comment]
All suggestions accepted

File: userguide/tutorials/metrics-newrelic.adoc [hunk 16]
```diff
@@ -82,38 +82,55 @@ installing the *New Relic Agent* in the system producing the data.
...
 
 To install the agent: 
 
-1. Login to your EC2 instance. 
-2. Paste the saved install command to the Ubuntu command line. 
-3. Press Enter. The installation should begin.
+[start=6]
+.. Login to your EC2 instance. 
+.. Paste the saved install command to the Ubuntu command line. 
+.. Press Enter. The installation should begin.
```
Comment (reshmabidikar) [inline userguide/tutorials/metrics-newrelic.adoc:98]: Numbering begins with `f`. Are these sub-points within point 5? If so, shouldn't they begin with `a`?

File: userguide/tutorials/metrics-newrelic.adoc [hunk 17]
```diff
@@ -82,38 +82,55 @@ installing the *New Relic Agent* in the system producing the data.
...
 
-You will be presented with a list of options to install, as shown below. An additional option, Infrastructure Agent, is always installed. Press Enter to select all of the options.
+You will be presented with a list of options to install, as shown below. An additional option, Infrastructure Agent, is always installed.
+
+[start=7]
+. Press Enter to select all of the options.
```
Comment (reshmabidikar) [inline userguide/tutorials/metrics-newrelic.adoc:103]: Again numbering does not look right. After `5`, the next number is `7`, so `6` is missing.

File: userguide/tutorials/metrics-newrelic.adoc [hunk 18]
```diff
@@ -82,38 +82,55 @@ installing the *New Relic Agent* in the system producing the data.
...
+*Note:* If you are using the multi-tier installation, you will have multiple EC2
 instances. The agent should be installed separately for each one.
 
-To test the installation, return to the main page. Select *Browse Data* from the top menu, then select *Metrics* from the dropdown.
+== Test the Installation
+
+. Return to the main page. 
+. Select *Browse Data* from the top menu. 
+. Select *Metrics* from the dropdown.
+
```
Comment (reshmabidikar) [inline userguide/tutorials/metrics-newrelic.adoc:129]: It's a bit unclear which "main page" is being referred to here. Is this the main page on the `New Relic` website?

File: userguide/tutorials/metrics-newrelic.adoc (hunk 16 above)
Comment (ghost) [reply userguide/tutorials/metrics-newrelic.adoc:98]: Right above it is [start=6]. If you remove that, it should reset to "a."

**Review by ghost — APPROVED** [review]

File: userguide/tutorials/metrics-newrelic.adoc (hunk 16 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:98]: fixed

File: userguide/tutorials/metrics-newrelic.adoc (hunk 17 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:103]: fixed

File: userguide/tutorials/metrics-newrelic.adoc (hunk 18 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:129]: Yes. Is there a clearer way to write this?

File: userguide/tutorials/metrics-newrelic.adoc (hunk 16 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:98]: fixed

File: userguide/tutorials/metrics-newrelic.adoc (hunk 17 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:103]: fixed

File: userguide/tutorials/metrics-newrelic.adoc (hunk 18 above)
Comment (jdmooney) [reply userguide/tutorials/metrics-newrelic.adoc:129]: clarified

**Review by reshmabidikar — APPROVED** [review]

**Pull request merged**